pkg/encryption/encryption.go

```diff
@@ -0,0 +1,187 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	b64 "encoding/base64"
+	"encoding/hex"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/dapr/components-contrib/secretstores"
+	"github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+)
+
+type Algorithm string
+
+const (
+	primaryEncryptionKey   = "primaryEncryptionKey"
+	secondaryEncryptionKey = "secondaryEncryptionKey"
+	errPrefix              = "failed to extract encryption key"
+	AES256Algorithm        = "AES256"
+	FeatureName            = "encryption"
```

Comment:
nit: please, use this enum instead: https://github.com/dapr/dapr/blob/a30488d55fa493f95ed3430539d7a348a96e6e22/pkg/config/configuration.go#L40

Reply:
Also, name it State.Encryption.

Reply:
That's the first time I see the "centralized" list.. it's mostly an anti-pattern in Go as packages are expected to be self-governed and all encompassing, also test dependencies or 3rd parties wanting to use the feature names now need to pull the dependencies of configuration. I made that change for now but will probably open an issue to disperse these among their respective packages.